Creating a first global layout

EmileCalixte opened the pull request:
This pull request aims to create a first layout for the app, using Bootstrap and custom CSS rules.

EmileCalixte committed:
Edit Style Of Footer

Edited style of footer in main layout

EmileCalixte committed:
Switch Order Of Jquery And Bootstrap Script Import

Imported Jquery before bootstrap javascript in main layout

EmileCalixte committed:
Fix Footer Paragraph Margin Bottom

Removed margin-bottom from the paragraph in footer, it caused a
one-pixel scroll on the page.

EmileCalixte committed:
Create Navbar component

Created a flexible navbar component, initialized in MainController and
used to display the header navbar in main layout

EmileCalixte committed:
Add Navbar initialization in MainController

Added HEADER_NAVBAR_CONFIG constant in MainController: array
representing navbar configuration
Added navbar initialization in MainController::beforeAction

EmileCalixte committed:
Just a save - working on bootstrap navbar

EmileCalixte committed:
Fix error when user is guest

Fixed an error when constructing Navbar object in case the user is guest

EmileCalixte committed:
Add getHTML functions in Navbar component

Added getHTML() functions in Navbar, NavbarChildItem, NavbarMainItem
and NavbarItemDropdownDivider classes

EmileCalixte committed:
Fix Navbar structure constant in MainController

The url for the proposals items was not correct

EmileCalixte committed:
Added Navbar Component In Main Layout

The navbar displayed in the main layout is now the navbar component

EmileCalixte committed:
Resolved conflict in views/layouts/main.php

EmileCalixte committed:
Fix Duplicate Title In Navbar Dropdown Item

EmileCalixte committed:
Merge branch 'master' into pre-layout

EmileCalixte committed:
Add navbar elements

Added elements to navbar

EmileCalixte committed:
Merge branch 'master' into pre-layout

EmileCalixte committed:
Add Create Account Item In Navbar

Added Create Account item in navbar, under Manage item

alexisljn commented on components/view/navbar/Navbar.php:
```diff
@@ -0,0 +1,94 @@
+     *                      'action-name',
+     *                      'action-name',
+     *                      // OR
+     *                      '*'
```
This should be an array OR a '*' but not both.

EmileCalixte replied:
Well regarded !

alexisljn commented on components/view/navbar/Navbar.php:
```diff
@@ -0,0 +1,94 @@
+     *              ],
+     *              [
+     *                  ...
+     *              ]
```
Indentation errors for others main items

EmileCalixte replied:
Yup, will fix it

alexisljn commented on views/layouts/main.php:
```diff
@@ -35,40 +36,22 @@
+    </header>
 
     <div class="container">
         <?= Breadcrumbs::widget([
```
We don't use the Breadcrumbs widget

EmileCalixte replied:
Yes, I'll remove it

EmileCalixte committed:
Fixed some errors

Removed breadcrumbs widget from main layout and fixed indentation errors
in Navbar constructor PHPDoc

alexisljn reviewed (approved):
Nothing to add
